gcp-observability/build.gradle

```diff
@@ -28,16 +28,15 @@ dependencies {
     implementation project(':grpc-protobuf'),
             project(':grpc-stub'),
             project(':grpc-alts'),
+            libraries.google_auth_credentials,
             libraries.google_auth_oauth2_http,
             libraries.autovalue_annotation,
             libraries.perfmark,
             ('com.google.guava:guava:31.0.1-jre'),
             ('com.google.errorprone:error_prone_annotations:2.11.0'),
-            ('com.google.auth:google-auth-library-credentials:1.4.0'),
             ('org.checkerframework:checker-qual:3.20.0'),
-            ('com.google.auto.value:auto-value-annotations:1.9'),
```

Comment:
@DNVindhya this was added because we were/are using AutoValue in o11y right? Looks like we stopped using it but didn't remove this dependency?

Reply:
This was initially added as an explicit dependency to fix version conflicts caused by `google-cloud-logging` library. 

When we started using `AutoValue`, `libraries.autovalue_annotation` was added as a dependency. 
You are right, we missed removing this dependency.

Reply:
Yeah, right now it is a duplicate dependency because of `libraries.autovalue_annotation`. I had to muck with the dependencies some (and then found a way to undo some of the changes), and that caused confusion.